R/gppt_graph.R

```diff
@@ -0,0 +1,266 @@
+gppt_region_graph <- gppt_final %>%
+  select(
+    region_code, region_name, question_number, summary_flag,
+    total_flag, question, answers, year, value, sort_order
+  ) %>%
+  filter(
+    region_code == "Y59", question_number == "Q01", summary_flag == FALSE,
+    total_flag == FALSE
+  ) %>%
+  group_by(region_code, region_name, question, question_number, answers, year, sort_order) %>%
+  summarise(value = sum(value))
+
+
+region_stack <- ggplot(gppt_region_graph, aes(x = year, y = value, fill = reorder(answers, sort_order))) +
+  geom_bar(
+    stat = "identity",
+    position = "stack"
+  ) +
+  geom_hline(yintercept = 0, size = 1, colour = "#333333") +
+  theme_scw() +
+  scale_fill_diverging(discrete = TRUE) +
+  scale_y_continuous(labels = scales::label_number_si()) +
+  labs(
+    title = gppt_region_graph$question,
+    subtitle = gppt_region_graph$region_name,
+    y = "Number of Patients",
+    x = "Year"
+  )
+region_stack
+
+region_perc <- ggplot(gppt_region_graph, aes(x = year, y = value, fill = reorder(answers, sort_order))) +
+  geom_bar(
+    stat = "identity",
+    position = "fill"
+  ) +
+  theme_scw() +
+  geom_hline(yintercept = 0, size = 1, colour = "#333333") +
+  scale_y_continuous(
+    name = "Percentage of Patients",
+    labels = scales::label_percent()
+  ) +
+  scale_fill_diverging(discrete = TRUE) +
+  labs(
+    title = gppt_region_graph$question,
+    subtitle = gppt_region_graph$region_name,
+    x = "Year"
+  )
+region_perc
+
+#### National Level Graphs
+
+gppt_national_graph <- gppt_final %>%
+  select(
+    question_number, summary_flag,
+    total_flag, question, answers, year, value, sort_order
+  ) %>%
+  filter(
+    question_number == "Q01", summary_flag == FALSE,
+    total_flag == FALSE
+  ) %>%
+  group_by(question, question_number, answers, year, sort_order) %>%
+  summarise(value = sum(value))
+
+
+national_stack <- ggplot(gppt_national_graph, aes(x = year, y = value, fill = reorder(answers, sort_order))) +
+  geom_bar(
+    stat = "identity",
+    position = "stack"
+  ) +
+  geom_hline(yintercept = 0, size = 1, colour = "#333333") +
+  theme_scw() +
+  scale_fill_diverging(discrete = TRUE) +
+  scale_y_continuous(labels = scales::label_number_si()) +
+  labs(
+    title = gppt_national_graph$question,
+    subtitle = "National Results",
+    y = "Number of Patients",
+    x = "Year"
+  )
+national_stack
+
+national_perc <- ggplot(gppt_national_graph, aes(x = year, y = value, fill = reorder(answers, sort_order))) +
+  geom_bar(
+    stat = "identity",
+    position = "fill"
+  ) +
+  theme_scw() +
+  geom_hline(yintercept = 0, size = 1, colour = "#333333") +
+  scale_y_continuous(
+    name = "Percentage of Patients",
+    labels = scales::label_percent()
+  ) +
+  scale_fill_diverging(discrete = TRUE) +
+  labs(
+    title = gppt_national_graph$question,
+    subtitle = "National Results",
+    x = "Year"
+  )
+national_perc
```

Comment:
It looks like the plots for each level are more or less the same, bar some small changes specific to the national-level? I don't think it's necessary to repeat all this.

Reply:
The individual levels are grouping by different things